### DISCUSSION TITLE
Doubt on Connection Packet offset value

### DISCUSSION THREAD

**Discussion by charankamarapu** [oracle/python-oracledb#202] [Q&A] [answered]
In Connection packet which is sent at the starting for connection, why is the offset for connection data is [74](https://github.com/oracle/python-oracledb/blob/b82db4c434d58fa5f3f811fa3c2a1e1672804638/src/oracledb/impl/thin/messages.pyx#L1778C4-L1778C4) shouldn't it be 70 ? From the code I can see connection data start after 70th byte.

**Comment by anthony-tuininga** [comment, marked as answer]
What makes you think it should be 70? I just looked at the packet output and it looks indeed like the correct value is 74. This is the beginning of the first packet sent (which you can see by setting the environment variable `PYO_DEBUG_PACKETS` to any value before running your program):

```
0000 : 01 14 00 00 01 00 00 00 |........|
0008 : 01 3F 01 2C 04 01 20 00 |.?.,....|
0016 : FF FF 4F 98 00 00 00 01 |..O.....|
0024 : 00 CA 00 4A 00 00 00 00 |...J....|
0032 : 84 84 00 00 00 00 00 00 |........|
0040 : 00 00 00 00 00 00 00 00 |........|
0048 : 00 00 00 00 00 00 00 00 |........|
0056 : 00 00 00 00 20 00 00 00 |........|
0064 : FF FF 00 00 00 00 00 00 |........|
0072 : 00 01 28 44 45 53 43 52 |..(DESCR|
0080 : 49 50 54 49 4F 4E 3D 28 |IPTION=(|
0088 : 41 44 44 52 45 53 53 3D |ADDRESS=|
0096 : 28 50 52 4F 54 4F 43 4F |(PROTOCO|
```
